File: 99-Continue/160/LeftHander.py
from collections import deque

class Node:
    def __init__(self,data:int):
        self.left = self.right = None
        self.data = data

class BinaryTree:
    def __init__(self):
        self.root = None
        
    def buildTree(self,arr:list[str])-> Node:
        if len(arr)==0:
            return self.root
        
        self.root = Node(int(arr[0]))
        queue = deque([self.root])
        i = 1
        while queue and i<len(arr):
            current = queue.popleft()
            if i<len(arr):
                if arr[i]!="null":
                    current.left= Node(int(arr[i]))
                    queue.append(current.left)
                i+=1
            if i<len(arr):
                if arr[i]!="null":
                    current.right = Node(int(arr[i]))
                    queue.append(current.right)
                i+=1
        return self.root
        
    def sumLefter(self)->int:
        if self.root is None:
            return 0
        tot=0
        queue = deque([self.root])
        while queue:
            current = queue.popleft()
            if current.left is not None:
                queue.append(current.left)
                if current.left.left is None and current.left.right is None:
                    tot+=current.left.data
            if current.right is not None:
                queue.append(current.right)
        return tot
        
def main():
    n = int(input())
    tree = BinaryTree()
    s = input().split()
    tree.buildTree(s)
    print(tree.sumLefter())
    
if __name__=="__main__":
    main()

File: 99-Continue/160/LeftHander.java
import java.util.*;

public class LeftHander {

    public static void main(String[] args) {
       Scanner sc = new Scanner(System.in);
       int n = sc.nextInt();
       sc.nextLine();
       String s = sc.nextLine();
       String[] strarr = s.split(" ");
       BinaryTree tree = new BinaryTree();
       tree.buildTree(strarr);
       System.out.println(tree.sumLefter());
       sc.close();
    }
}

class Node{
    int data;
    Node left,right;
    Node(int data){
        this.data=data;
        this.left=this.right=null;
    }
}

class BinaryTree{
    Node root;
    BinaryTree(){
        this.root=null;
    }
    
    Node buildTree(String[] arr){
        if(arr.length==0){
            return root;
        }
        Queue<Node> queue = new LinkedList<>();
        root = new Node(Integer.parseInt(arr[0]));
        queue.offer(root);
        int i=1;
        while(!queue.isEmpty()&&i<arr.length){
            Node current = queue.poll();
            if(i<arr.length){
                if(!arr[i].equals("null")){
                    current.left = new Node(Integer.parseInt(arr[i]));
                    queue.offer(current.left);
                }
                i++;
            }
            if(i<arr.length){
                if(!arr[i].equals("null")){
                    current.right = new Node(Integer.parseInt(arr[i]));
                    queue.offer(current.right);
                }
                i++;
            }
        }
        return root;
    }
    
    int sumLefter(){

        int sum=0;
        Queue<Node> queue = new LinkedList<>();
        queue.offer(root);
        while(!queue.isEmpty()){
            Node current = queue.poll();
            if(current.left!=null){
                queue.offer(current.left);
                if(current.left.left==null && current.left.right==null){
                    sum+=current.left.data;
                }
            }
            if(current.right!=null){
                queue.offer(current.right);
               
            }
        }
        return sum;
    }
}